online-shop/src/app/components/presentational/product-details-view/product-details-view.component.html

```diff
@@ -1,29 +1,29 @@
 <div class="wrapper">
   <div class="title-flex">
-    <h1>Product: {{ product.name }}</h1>
+    <h1 *ngIf="product">Product: {{ product.name }}</h1>
     <div class="buttons">
-      <button class="edit-button">Edit</button>
-      <button class="delete-button">Delete</button>
+      <button *ngIf="product" [routerLink]="['/edit',product.id]" class="edit-button" >Edit</button>
+      <button *ngIf="product" class="delete-button" (click)="onDeleteProduct(product.id)" [routerLink]="['']">Delete</button>
     </div>
   </div>
 
   <div class="details">
     <div class="left-details">
       <h2>Details</h2>
-      <p class="name"><strong>Name:</strong> {{ product.name }}</p>
-      <p class="category"><strong>Category:</strong> {{ product.category }}</p>
-      <p class="price"><strong>Price:</strong> {{ product.price }} Ron</p>
+      <p *ngIf="product" class="name"><strong>Name:</strong> {{ product.name }}</p>
```

Comment:
Instead of having so many ngIfs which check the existence of the product you could have only one check placed at an upper place in the DOM (for example you could place it in <div class="title-flex"> and get rid of all the ngIfs in the child elements)

Reply:
Modified. Less ngIfs :D